### ISSUE TITLE
Figure out how to loosen selinux policy for openvpn

### ISSUE THREAD

**Issue by zenhack** [CCI-MOC/hil-vpn#11]
The design of hil-vpn involves running several instances of openvpn-server, each listening on a different port. It seems the default selinux policy denies openvpn acess to all but the standard openvpn port. We should work out how to adjust the policy on a local system so we can get this working without just disabling selinux.

**Comment by naved001** [comment]
I installed policycoreutils-python on centos, and then allowed custom ports for openvpn.
`semanage port -a -t openvpn_port_t -p udp 6970`
```
[root@localhost ~]# semanage port -l |grep openvpn
openvpn_port_t                 tcp      1194
openvpn_port_t                 udp      6970, 1194
```
I was then able to run an openvpn process on the non standard port. (This was with SELinux set to enforcing). Running it on other non standard port failed (as expected).

We would probably want a range and that can be specified like so
```
[root@localhost ~]# semanage port -a -t openvpn_port_t -p udp 6971-6980
[root@localhost ~]# semanage port -l |grep openvpn
openvpn_port_t                 tcp      1194
openvpn_port_t                 udp      6971-6980, 6970, 1194
```

**Comment by zenhack** [comment]
Sounds good. We'll add this to some setup documentation when we get around to writing it, and then can consider this issue solved.

**Comment by zenhack** [comment]
@naved001, does that persist changes on boot?

**Comment by naved001** [comment]
Yes, just rebooted my test setup to check and the changes persisted.

**Comment by zenhack** [comment]
Working on getting more of the vagrant setup to be automatic, partially to make dev easier, but also so we can use it for integration testing (I'm not sure where we'll run it that way though, as it's not clear how we'd do it in travis).

In the course of doing so I added the above to the provisioning script, and found there are some other issues with SELinux; the one we bang into first is that openvpn is denied access to the up hook script. I suspect we'll hit a few more things before we're done.

**Comment by zenhack** [comment]
So it's here for reference: I sorted out the other issues, and the fixes are part of the Vagrantfile. Still need to move this stuff to documentation.

**Comment by zenhack** [comment]
Just a note, distro packages (#18) should be constructed to do this automatically.
